lib/takeaway.rb

```diff
@@ -0,0 +1,41 @@
+class Takeaway
+
+  attr_reader :current_order
+
+  def initialize(menu = Menu.new)
+    @dishes = menu.dishes
+    @current_order = {}
+    @total = 0
+  end
+
+  def show_menu
+    @dishes
+  end
+
+  def order(dish, quantity = 1)
+    raise 'Dish not available: not part of the menu' unless @dishes.include?(dish)
+    @current_order.merge!({ dish => quantity })
+    end
+
+  def show_basket
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for show_basket is too high. [32.57/15] (http://c2.com/cgi/wiki?AbcMetric)<br>Metrics/MethodLength: Method has too many lines. [15/10] (https://github.com/bbatsov/ruby-style-guide#short-methods)